extensions-contrib/opencensus-extensions/src/main/java/org/apache/druid/data/input/opencensus/protobuf/OpenCensusProtobufInputFormat.java

```diff
@@ -69,34 +66,24 @@ public boolean isSplittable()
   @Override
   public InputEntityReader createReader(InputRowSchema inputRowSchema, InputEntity source, File temporaryDirectory)
   {
-    try {
+    if (source instanceof KafkaRecordEntity) {
       KafkaRecordEntity kafkaInputEntity = (KafkaRecordEntity) source;
       Header versionHeader = kafkaInputEntity.getRecord().headers().lastHeader(VERSION_HEADER_KEY);
-      int version = 0;
       if (versionHeader != null) {
-        version = ByteBuffer.wrap(versionHeader.value()).order(ByteOrder.LITTLE_ENDIAN).getInt();
-      }
-
-      if (version == 1) {
-        return new OpenTelemetryMetricsProtobufReader(
-          inputRowSchema.getDimensionsSpec(),
-          (ByteEntity) source,
-          metricDimension,
-          DEFAULT_VALUE_DIMENSION,
-          metricLabelPrefix,
-          resourceLabelPrefix
-        );
+        int version =
+          ByteBuffer.wrap(versionHeader.value()).order(ByteOrder.LITTLE_ENDIAN).getInt();
+        if (version == 1) {
```

Comment:
nit, we should declare a constant such as `OPENTELEMETRY_FORMAT` for this `1` value

Reply:
make sense, updated

Reply:
make sense! Updated with `OPENTELEMETRY_FORMAT_VERSION`